### ISSUE TITLE
Passing incorrect type to function does not produce compiler errors 

### ISSUE THREAD

**Issue by jamcoupe** [Microsoft/TypeScript#22117]
**TypeScript Version:**  
* Version 2.7.2
* Version 2.8.0-dev.20180222

**Search Terms:** 
* typed Function parameter accepts any type
* function accepts all types
* Passing incorrect type to function does not produce compiler errors 

**Code**
tsconfig.json
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es5",
    "strict": true,
    "sourceMap": true
  },
  "exclude": [
    "node_modules"
  ]
}
```

index.ts
```ts
export class A {}
export class B {}

export function typedFunction(item: A): void {}

const a = new A();
const b = new B();

typedFunction(a);
typedFunction(b);
typedFunction('string');
typedFunction(123);
typedFunction(true);
```
**Expected behavior:**

Errors from the compiler for trying to pass in types B, string, number and boolean.

**Actual behavior:**

No errors are output by the compiler.

**Playground Link:**  
[Playground](http://www.typescriptlang.org/play/index.html#src=export%20class%20A%20%7B%7D%0D%0Aexport%20class%20B%20%7B%7D%0D%0A%0D%0Aexport%20function%20typedFunction(item%3A%20A)%3A%20void%20%7B%0D%0A%20%20console.log(item)%3B%0D%0A%7D%0D%0A%0D%0Aconst%20a%20%3D%20new%20A()%3B%0D%0Aconst%20b%20%3D%20new%20B()%3B%0D%0A%0D%0AtypedFunction(a)%3B%0D%0AtypedFunction(b)%3B%0D%0AtypedFunction('string')%3B%0D%0AtypedFunction(123)%3B%0D%0AtypedFunction(true)%3B)

**Related Issues:**

**Comment by andy-ms** [comment]
TypeScript has structural types, so a class with no members will be matched by anything. Marking as duplicate of #202.
